### PR TITLE
feat: AtClientImpl for GET and PUT byte[] Method

### LINKED ISSUE #38: Implement `_getBinary(AtKey)` and `_put(AtKey, byte[])`
## _getBinary(AtKey) implementation
- [ ] `_getBinary(SharedKey, byte[])`
- [ ] `_getBinary(SelfKey, byte[])`
- [ ] `_getBinary(PublicKey, byte[])`

## _put(AtKey, byte[]) implementation
- [ ] `_put(SharedKey, byte[])` in `AtClientImpl.java`
- [ ] `_put(SelfKey, byte[])` in `AtClientImpl.java`
- [ ] `_put(PublicKey, byte[])` in `AtClientImpl.java`

### PR DESCRIPTION
**Pull request by JayUpadhyay-8**
closes #38

### CODE REVIEW

File: at_client/src/main/java/org/atsign/client/api/impl/clients/AtClientImpl.java [hunk 1]
```diff
@@ -580,12 +580,64 @@ private String _delete(PublicKey key) throws AtException {
...
+    }
 
     private String _put(SharedKey sharedKey, byte[] value) throws AtException {throw new RuntimeException("Not Implemented");}
     private String _put(SelfKey selfKey, byte[] value) throws AtException {throw new RuntimeException("Not Implemented");}
-    private String _put(PublicKey publicKey, byte[] value) throws AtException {throw new RuntimeException("Not Implemented");}
+    
+    private String _put(PublicKey publicKey, byte[] value) throws AtException {
+    
+        // 1. generate dataSignature for byte[]
+        publicKey.metadata.dataSignature = generateSignatureByte(value);
```
Comment (JeremyTubongbanua) [inline at_client/src/main/java/org/atsign/client/api/impl/clients/AtClientImpl.java:628]: I think for now, we should comment this line out. Not sure if we are generating the data signature correctly. I think we will re-implement this when we decide to focus more on data signatures in the metadata.

**Review by JeremyTubongbanua — CHANGES_REQUESTED** [review]
Looks great so far!

Could you provide some code and output to show it is working? Just send the code here in the conversation and some screenshots of the output.

Thanks a ton!

File: at_client/src/main/java/org/atsign/client/api/impl/clients/AtClientImpl.java (hunk 1 above)
Comment (JeremyTubongbanua) [reply at_client/src/main/java/org/atsign/client/api/impl/clients/AtClientImpl.java:628]: Actually, scratch that, please leave that in for now. I think this is fine :)

Great work!

**Comment by JayUpadhyay-8** [comment]
Hi Jeremy,

I have written a TempTestDriver class to test this code. I am providing its code and screenshot for you to better understand its working. 


`
public class TemporaryTest {
    public static void main(String[] args) throws Exception {
        AtSign atSign = new AtSign("@16ourfeminist");
        AtClient atClient = AtClient.withRemoteSecondary("root.atsign.org:64", atSign, true);

        PublicKey pk = new KeyBuilders.PublicKeyBuilder(atSign).key("test").build();

        byte[] value = "testToken".getBytes();

        String response = atClient.put(pk, value).get();

        System.out.println("response here -- " + response);

        byte[] getPublicValue = atClient.getBinary(pk).get();

        String str = new String(getPublicValue);

        System.out.println("get public key -- " + str + " get public value -- " + getPublicValue);

    }

}
`
TempTest Output

**Comment by JeremyTubongbanua** [comment]
Closing, please make a PR to this [branch](https://github.com/atsign-foundation/at_java/tree/byte_at_client_operations)

**Comment by JeremyTubongbanua** [comment]
Comment from @gkc :

Please utilize Base 2e15 encoding/decoding

The encoding/decoding should be compatible with our Dart client

https://github.com/atsign-foundation/at_client_sdk/blob/trunk/packages/at_client/lib/src/client/at_client_impl.dart#L447-L449

https://github.com/atsign-foundation/at_client_sdk/blob/trunk/packages/at_client/lib/src/transformer/response_transformer/get_response_transformer.dart#L62-L69

Feel free to add any thoughts @gkc

**Pull request closed without merging**